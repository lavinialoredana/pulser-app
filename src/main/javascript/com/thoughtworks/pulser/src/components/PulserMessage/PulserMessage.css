.Pulser-message-container{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    margin-right: 3rem;
}

.Title-image-reaction-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
    width: calc(100% / 2);
    height: 400px;
    margin: 0 3rem;
}

.Reaction-image {
    height: 200px;
    width: 200px;
    margin-top: 2rem;
    display: flex;
    flex-direction: column;
}