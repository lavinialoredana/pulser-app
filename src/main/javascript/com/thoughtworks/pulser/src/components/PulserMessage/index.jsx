import React from "react";

const PulserMessage = ()=>{
    return(<div className="Pulser-message">
     <div data-testid="reaction-title">
      <span> Happy</span>
     </div>
    </div>)
};

export default PulserMessage;