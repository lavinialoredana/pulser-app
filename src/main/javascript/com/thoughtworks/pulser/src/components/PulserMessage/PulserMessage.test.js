import { cleanup, render, screen } from "@testing-library/react";
import PulserMessage from "./index.jsx";

describe("PulserMessage component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<PulserMessage />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should contain a reaction title", ()=>{
    render(<PulserMessage/>);

    const titleElement = screen.getByTestId("reaction-title");
    const titleText = screen.getByText("Happy")

    expect(titleElement).toBeInTheDocument();
    expect(titleText).toBeInTheDocument()
  })
});
