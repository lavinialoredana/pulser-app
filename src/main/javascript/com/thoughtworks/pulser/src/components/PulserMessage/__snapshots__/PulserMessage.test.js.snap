// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PulserMessage component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Pulser-message"
  >
    <div
      data-testid="reaction-title"
    >
      <span>
         Happy
      </span>
    </div>
  </div>
</DocumentFragment>
`;
